hugophotoswipe/photo.py

```diff
@@ -105,13 +108,105 @@ def _load_original_image(self):
         # fallback for unhandled rotation tags
         return img
 
+    def _load_exif(self, image):
+        if settings.exif:
+            tags = set(_filter_tags(TAGS.values(),
+                                    settings.exif.get('include'),
+                                    settings.exif.get('exclude')))
+            tags.add('Orientation')  # always need this for resize
+        else:
+            tags = TAGS.values()
+
+        exif_data = {}
+        exif = image._getexif() or {}
+        for k, v in exif.items():
+            decoded = TAGS.get(k)
+            if decoded in tags and not isinstance(v, IFDRational):  # Filter complex data values
+                exif_data[decoded] = v
+        for k, v in exif_data.pop('GPSInfo', {}).items():
+            decoded = GPSTAGS.get(k, k)
+            exif_data[decoded] = v
+        self._exif = exif_data
+
     def free(self):
         """Manually clean up the cached image"""
         if hasattr(self, "_original_img") and self._original_img:
             self._original_img.close()
             del self._original_img
         self._original_img = None
 
+    @property
+    def iptc(self):
+        if self._iptc is None:
+            if settings.iptc:
+                tags = _filter_tags(iptc3.c_datasets_r.keys(),
+                                    settings.iptc.get('include'),
+                                    settings.iptc.get('exclude'))
+            else:
+                tags = iptc3.c_datasets_r.keys()
+
+            info = iptc3.IPTCInfo(self.original_path)
+            iptc = {}
+            for k in tags:
+                if type(info[k]) is bytes:
+                    iptc[k] = info[k].decode('utf-8')
+                elif type(info[k]) is list:
+                    l = []
+                    for v in info[k]:
+                        l.append(v.decode('utf-8'))
+                    iptc[k] = l
+                else:
+                    iptc[k] = info[k]
+            self._iptc = iptc
+
+        return self._iptc
+
+    @property
+    def exif(self):
+        if not self._exif:
+            _ = self.original_image  # Trigger loading image and exif data
+        return self._exif
+
+    def _get_tag_value(self, tag):
+        assert tag is not None
+        try:
+            obj, t = tag.split('.')
+        except Exception as e:
```

Comment:
I think the only error you can get here is a `ValueError`, right? Then you can just catch that instead of the general `Exception`.

Reply:
Improved. Also added tests to check that malformed tag maps do indeed throw the expected error.